Bash/InstallHuntress-macOS-bash.sh

```diff
@@ -103,49 +125,84 @@ Usage: $0 [options...] --account_key=<account_key> --organization_key=<organizat
-  fi
-  case "$OPT" in
-    a | account_key)
-        account_key="$OPTARG"
-        ;;
-    o | organization_key)
-        organization_key="$OPTARG"
-        ;;
-    t | tags)
-        tags="$OPTARG"
-        ;;
-    i | install_system_extension)
-        install_system_extension=true
-        ;;
-    h | help)
-        usage
-        ;;
-    ??*)
-        logger "Illegal option --$OPT"
-        exit 2
-        ;;  # bad long option
-    \? )
-        exit 2
-        ;;  # bad short option (error reported via getopts)
-  esac
+reinstall=false
+while getopts "a:o:t:ihr-:" OPT; do
+    if [ "$OPT" = "-" ]; then
+        OPT="${OPTARG%%=*}"       # extract long option name
+        OPTARG="${OPTARG#$OPT}"   # extract long option argument (may be empty)
+        OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
+    else
+        # the user used a short option, but we still want to strip the assigning `=`
+        OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
+    fi
+    case "$OPT" in
+        a | account_key)
+            account_key="$OPTARG"
+            ;;
+        o | organization_key)
+            organization_key="$OPTARG"
+            ;;
+        t | tags)
+            tags="$OPTARG"
+            ;;
+        i | install_system_extension)
+            logger "Running with System Extension immediate install option"
+            install_system_extension=true
+            ;;
+        r | reinstall)
+            logger "Running with the -reinstall flag"
+            reinstall=true
+            ;;
+        h | help)
+            usage
+            copyLogAndExit "0"
+            ;;
+        ??*)
+            logger "Illegal option --$OPT"
+                copyLogAndExit "2"
+            ;;  # bad long option
+        \? )
+                copyLogAndExit "2"
+            ;;  # bad short option (error reported via getopts)
+    esac
 done
 shift $((OPTIND-1)) # remove parsed options and args from $@ list
 
+# try/catch, if the connectivity tester fails to execute we'll log that as an error.
+for hostn in "update.huntress.io" "huntress.io" "eetee.huntress.io" "huntress-installers.s3.amazonaws.com" "huntress-updates.s3.amazonaws.com" "huntress-uploads.s3.us-west-2.amazonaws.com" "huntress-user-uploads.s3.amazonaws.com" "huntress-rio.s3.amazonaws.com" "huntress-survey-results.s3.amazonaws.com"; do 
+    logger "$(nc -z -v $hostn 443 2>&1)" || (logger "error occured during network connectivity test")
+done
+
+# Check for existing Huntress install, if already installed exit with error. Bypass if using the reinstall flag.
+if [ $reinstall = false ]; then
+    if [ -d "/Applications/Huntress.app/contents/macos" ]; then
+        logger "Huntress assets found, checking for running processes"
+        numServicesStopped=0
+        for HuntressProcess in "HuntressAgent" "HuntressUpdater"; do
+            if [ $(pgrep $HuntressProcess > /dev/null) ]; then
```

Comment:
```suggestion
            if [ $(pgrep "$HuntressProcess" > /dev/null) ]; then
```

Always wrap string expansion in quotes